### PR TITLE
Add credits client

### PR DESCRIPTION
**Pull request by ray-kast**
- Added a TOML parser for the credits client.
- Add producer to credits client, minus schema.

### CODE REVIEW

File: crates/core/src/credits.rs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            .credit_sheet
+            .get(&item)
+            .ok_or_else(|| anyhow!("No price associated with credit line item {item:?}"))?;
+
+        self.producer
+            .send(Some(&CreditsEvent {}), Some(&CreditsEventKey {}))
+            .await
+            .map_err(Into::into)
+    }
+}
```
Comment (kespinola) [inline crates/core/src/credits.rs:92]: Need to send the user id of the initiator and the organization id when sending a deduction.

Also recommend calling it `send_deduction`

**Review by kespinola — CHANGES_REQUESTED** [review]
I don't see api for reading the credit sheet. Did I miss or item to build?

File: crates/core/src/credits.rs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        I: Borrow<Q>,
+    {
+        let credits = self.get_cost(item)?;
+
+        self.producer
+            .send(Some(&CreditsEvent {}), Some(&CreditsEventKey {}))
+            .await
+            .map_err(Into::into)
+    }
+}
```
Comment (kespinola) [inline crates/core/src/credits.rs:113]: `credits` isn't getting sent with the event. And maybe we have the caller pass the CreditEventKey { organization_id, user_id }

File: crates/core/src/credits.rs [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+use crate::{prelude::*, producer, util::DebugShim};
+
+impl producer::Message for CreditsEvent {
+    type Key = CreditsEventKey;
+}
+
+/// Service startup configuration for charging credits and reading the credit
+/// sheet
+#[derive(Debug)]
```
Comment (imabdulbasit) [inline crates/core/src/credits.rs:13]:
```suggestion
#[derive(Debug, Clone)]
```
need the credits client to implement Clone

File: crates/core/src/credits.rs [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+impl Config {
+    /// Construct a new credits producer client from this config instance
+    #[inline]
+    pub async fn build<I: LineItem>(self) -> Result<CreditsClient<I>> {
+        CreditsClient::new(self).await
+    }
+}
+
+/// A client for producing credit deduction line items
+#[derive(Debug)]
```
Comment (imabdulbasit) [inline crates/core/src/credits.rs:28]:
```suggestion
#[derive(Debug, Clone)]
```
need the credits client to implement Clone

File: crates/core/Cargo.toml [hunk 5]
```diff
@@ -12,7 +12,9 @@ keywords = ["solana", "holaplex", "web3"]
 categories = ["cryptography::cryptocurrencies", "web-programming"]
 
 [features]
-kafka = ["rdkafka"]
+credits = ["kafka_internal", "strum", "toml"]
```
Comment (imabdulbasit) [inline crates/core/Cargo.toml:15]:
```suggestion
credits = ["kafka_internal", "strum", "toml", "rand"]
```

**Review by imabdulbasit — APPROVED** [review]

**Pull request merged**